tests/cursor/jsfmt.spec.js

```diff
@@ -36,3 +63,124 @@ foo('bar', cb => {
     cursorOffset: 23
   });
 });
+
+test("works when the file starts with a comment", () => {
+  expect(
+    runPrettierWithInlineCursor(` 
+    // hi<|> lol
+    haha() 
+`)
+  ).toMatchSnapshot();
+
+  expect(
+    runPrettierWithInlineCursor(` 
+    // hi lol
+    haha()<|>
+`)
+  ).toMatchSnapshot();
+});
+
+test("puts the cursor in sensible places", () => {
+  expect(runPrettierWithInlineCursor(`return        <|> 15`)).toMatchSnapshot();
+  expect(runPrettierWithInlineCursor(`return        <|>15`)).toMatchSnapshot();
+  expect(
+    runPrettierWithInlineCursor(`
+foo  <|>  (bar);  
+`)
+  ).toMatchSnapshot();
+
+  expect(
+    runPrettierWithInlineCursor(` 
+
+  <|>  
+    
+
+  const y = 5
+`)
+  ).toMatchSnapshot();
+
+  expect(
+    runPrettierWithInlineCursor(` 
+
+  const y = 5
+
+     
+  
+  <|>  
+    
+
+
+  const z = 9
+`)
```

Comment:
What about

```js
const y = 5
<|>



const z = 9
```

IMO that should either put it on the blank line between the two or right after the `5`.

Reply:
Yeah it puts it on the blank line between the two. I added this test case just now :+1: